### ISSUE TITLE
Create GitHub repo and enable Travis from CLI

### ISSUE THREAD

**Issue by jamiebuilds** [sindresorhus/generator-nm#28]
It would be nice if `yo nm` had the option of creating a GitHub repo, and enabling Travis on it. With the `hub` and `travis` CLIs this is fairly easy to do:

``` sh
$ git init my-repo
$ cd my-repo
$ yo nm
$ hub create -d 'My awesome repo'
$ travis enable

# code...

$ git add -A
$ git commit -m 'Init commit'
$ git push origin master
$ npm version major
$ npm publish
$ git push origin master --follow-tags
```

I put [this](https://github.com/thejameskyle/hub-bin) together, I'm trying to find a good way of packaging up the [travis gem](https://github.com/travis-ci/travis.rb) as well.

**Comment by jamestalmage** [comment]
I think I am :-1: on this.

In my mind there are only two lines of interest.

```sh
$ hub create -d 'My awesome repo'
$ travis enable
```

I **definitely** do not want the rest of those lines happening automatically.

Even then, I usually do not run `hub create` until I am sure I am onto something useful. I make a fair number of false starts that I quickly discard.

It seems like it would be better to create a wrapper-cli that does this:

```sh
$ hub-create -d 'My awesome repo'

# performs the following
git init  #if required
hub create -d 'My awsome repo'
travis enable
# set description in package.json
# set description line in README.md (that might be hard to parse).
```

**Comment by thejameskyle** [comment]
> I definitely do not want the rest of those lines happening automatically.

sorry, i just posted my entire copy-paste script

**Comment by sindresorhus** [comment]
I personally prefer to work in private until I have something worth showing and then do the publish travis dance.

**Comment by jamestalmage** [comment]
@sindresorhus what about a sub-generator (`nm:publish`) that handled the "publish travis dance".

You could defer most of the prompts until the latter steps.

```sh
$ yo nm
? Do you need a CLI (y/N)

yo nm:publish
? Init GitHub Repository (Y/n) #skipped if `git init` already ran and has at least one remote set.
? What do you want to name your module #sets name in package.json, and repo name for `hub create`
? Provide a description # sets description in package.json / Readme.md / and description for `hub create` 
? What is your GitHub username # defaults to package.json, followed by stored previous answer
? What is the URL of your website # defaults to value in package.json, followed by stored previous answer
? Enable Travis (Y/n)
 ```

I'd prefer to still `git push` on my own, so `nm:publish` may not be the correct name.

**Comment by sindresorhus** [comment]
I like that idea.

**Comment by perryprog** [comment]
My fork (#68) basically does this. Screenshot:

<img width="2004" alt="screen shot 2017-05-24 at 3 57 18 pm" src="https://cloud.githubusercontent.com/assets/12972285/26423300/25f8e7e4-409b-11e7-96e5-de786c306c56.png">


Also, I did not think of making this a sub generator... I'm wondering how important that is now?

**Comment by sindresorhus** [comment]
I'm going to pass on this. I don't want to maintain something I wouldn't use myself. I'm happy to link to alternative generators though.
